Main.hs

```diff
@@ -4,22 +4,39 @@ import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+-- short ls = length ls < 3
+short (a : b : c : xs) = False
+short _ = True
 
 lovely :: [Int] -> Bool
-lovely = undefined 
+lovely a = short a || a!!2 == 14
 
 rightTriangles :: [(Int, Int, Int)]
-rightTriangles = undefined 
+rightTriangles = [ (a, b, c) | c <- [1..], b <- [1..c], a <-[1..b], a^2 + b^2 == c^2]
 
 fizzBuzz :: [String]
-fizzBuzz = undefined 
+fizzBuzz = [s | i <- [1 .. ], let s | i `mod` 3 == 0 && i `mod` 5 == 0 = "FizzBuzz"
+                                    | i `mod` 3 == 0 = "Fizz"
+                                    | i `mod` 5 == 0 = "Buzz"
+                                    | otherwise = show i  ]
+
+toYears = 31557600
 
 ageOn :: String -> Float -> Float
-ageOn planet ageInSeconds = undefined 
+ageOn planet ageInSeconds | planet == "Mercury" = ageInSeconds /  0.2408467 / toYears
+                          | planet == "Venus" = ageInSeconds / 0.61519726 / toYears
+                          | planet == "Earth" = ageInSeconds / 1.0 / toYears
+                          | planet == "Mars" = ageInSeconds / 1.8808158 / toYears
+                          | planet == "Jupiter" = ageInSeconds / 11.862615 / toYears
+                          | planet == "Saturn" = ageInSeconds / 29.447498 / toYears
+                          | planet == "Uranus" = ageInSeconds / 84.016846 / toYears
+                          | planet == "Neptune" = ageInSeconds / 164.79132 / toYears
+                          | planet == "Pluto" = error "Pluto is not a planet"
```

Comment:
You have a lot of code duplication here. It's better to introduce a function for it.